luigi/contrib/lambda_runner.py

```diff
@@ -0,0 +1,141 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+AWS Lambda Runner for Luigi
+===========================
+
+Lambda is a serverless compute service from Amazon Web Services (AWS) that
+facilitates stateless, scalable code. These functions can be written in Python
+or a variety of other supported languages, and are easily interfaced with
+other AWS services.
+
+This module requires the `boto3` package to be installed.
+You must also have AWS credentials set up in way boto3 can detect them. See:
+https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html
+
+`LambdaRunnerTask` invokes a pre-existing Lambda function given a name/ARN.
+"""
+
+import logging
+import luigi
+import json
+import os
+import uuid
+
+logger = logging.getLogger('luigi-interface')
+
+try:
+    import boto3
+    client = boto3.client('lambda')
+except ImportError:
+    logger.warning('boto3 is required for lambda_runner but not installed')
+
+
+class LambdaRunnerTask(luigi.Task):
+    """
+    Base Task for invoking AWS Lambda functions. Must provide a Lambda name or
+    ARN. Optionally, specify input payload to the Lambda, which is passed as
+    event JSON to the invoked function. If no payload is specified, use an
+    empty JSON object. Payload can be a Python dictionary or filepath; either
+    way, it must be JSON seriablizable.
+
+    Writes the Lambda response payload as a plaintext JSON file using a
+    randomly generated uuid associated with the task. Stores the payload as the
+    `response_payload` field and other of the response metadata as `response`.
+
+    :param str lambda_name: Name or ARN of AWS Lambda function
+    :param str output_directory: Directory to write output to. Default is '.'
+    :param bool dry_run:
+        If True, invoke the Lambda function in dry_run mode, which will result
+        in an empty string payload written to output. Default is False
+    :param payload:
+        Object contents to pass as event JSON for the Lambda.
+        Cannot specify `filepath_payload` if using this. Default is None
+    :type payload: dict or None
+    :param filepath_payload:
+        Path for plaintext JSON file to use as event JSON for the Lambda.
+        Cannot specify `payload` if using this. Default is None
+    :type filepath_payload: str or None
+
+    :raises TypeError: if specified payload is not JSON serializable
+    """
+    # Parameters
+    lambda_name = luigi.Parameter()
+    output_directory = luigi.Parameter(default='.')
+    dry_run = luigi.BoolParameter(default=False)
+    payload = luigi.OptionalParameter(default=None)
+    filepath_payload = luigi.OptionalParameter(default=None)
+    # Fields set by the task
```

Comment:
Should these "Fields set by the task" really be class variables? Or should they be instance vars?